File: ios/Runner/Model/ArtistModel.swift
//
//  ArtistModel.swift
//  Runner
//
//  Created by t k on 2024/08/10.
//

import MediaPlayer

struct ArtistModel: ArtistModelContract {
    let media: MPMediaItemCollection

    var id: ArtistId {
        ArtistId(id: media.representativeItem?.artistPersistentID ?? 0)
    }

    var name: String {
        media.representativeItem?.artist ?? "undefined"
    }
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/model/ArtistModel.kt
package dev.tcode.thinmpf.model

import dev.tcode.thinmpf.model.contract.ArtistModelContract
import dev.tcode.thinmpf.model.valueObject.ArtistId

class ArtistModel(
    id: String,
    name: String
): ArtistModelContract {
    override val id: ArtistId = ArtistId(id)
    override val name: String = name
}
